omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+
+        obstacles = self._usd_help.get_obstacles_from_stage(
+            reference_prim_path=self.robot.root_link.prim_path,
+            ignore_substring=[
+                self.robot.prim_path,  # Don't include robot paths
+                "/World/ground_plane",  # Don't include collisions with floor
+                "visual",  # Don't include any visuals
+                "/curobo",  # Don't include curobo prim
+                *self._ignore_collision_paths,  # Don't include manually-specified collision paths
+                *self._ignore_paths,  # Don't include any additional specified paths
+            ],
+        ).get_collision_check_world()
+        self.mg.update_world(obstacles)
+        print("Synced CuRobo world from stage.")
+
+    def check_collisions(
+        self,
+        q,
+        activation_distance=0.01,
+    ):
+        """
+        Checks collisions between the sphere representation of the robot and the rest of the current scene
+
+        Args:
+            q (th.tensor): (B, N)-shaped tensor, representing B-total different joint configurations to check
+                collisions against the world
+            activation_distance (float): Safety buffer around robot mesh representation which will trigger a
+                collision check
+
+        Returns:
+            th.tensor: (B,)-shaped tensor, where each value is True if in collision, else False
+        """
+        # Update obstacles
+        self.update_obstacles()
+
+        # Compute kinematics to get corresponding sphere representation
+        cu_js = lazy.curobo.types.state.JointState(position=self.tensor_args.to_device(q))
+        robot_spheres = self.mg.compute_kinematics(cu_js).robot_spheres
+        # (N_samples, n_obs_spheres, 4) --> (N_samples, 1, n_spheres, 4)
+        robot_spheres = robot_spheres.unsqueeze(dim=1)
+
+        # Run direct collision check
+        with th.no_grad():
+            # Run the overlap check
+            # Sphere shape should be (N_queries, 1, n_obs_spheres, 4), where 4 --> (x,y,z,radius)
+            coll_query_buffer = lazy.curobo.geom.sdf.world.CollisionQueryBuffer()
+            coll_query_buffer.update_buffer_shape(
+                shape=robot_spheres.shape,
+                tensor_args=self.tensor_args,
+                collision_types=self.mg.world_coll_checker.collision_types,
+            )
+
+            dist = self.mg.world_coll_checker.get_sphere_collision(
+                robot_spheres,
+                coll_query_buffer,
+                weight=th.tensor([50000.0], device=self.tensor_args.device),
+                activation_distance=th.tensor([activation_distance], device=self.tensor_args.device),
+                env_query_idx=None,
+                return_loss=False,
+            ).squeeze(
+                dim=1
+            )  # shape (N_samples, n_spheres)
+
+            # Non-zero distances correspond to a collision detection (or close to a collision, within activation_distance
```

Comment:
does "non-zero" mean it can be positive or negative? In that case, should the logic be `equal_all(n_obs_spheres, 0)` instead?

Reply:
"positive" value, not "non-zero" value

Reply:
updated